src/twisted/internet/tcp.py

```diff
@@ -337,7 +337,18 @@ def getTcpNoDelay(self):
         return bool(self.socket.getsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY))
 
     def setTcpNoDelay(self, enabled):
-        self.socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, enabled)
+        try:
+            # There are bug reports about failures when setting TCP_NODELAY under certain conditions
```

Comment:
I am ok to have this fix here, but this looks like a different fix to the one described in the original bug report.

I think that we need to have a separate newsfragment file for this bugfix.